### PR TITLE
build: Automate dev env setup

### PR DESCRIPTION
**Pull request by armcknight**
Automates the setup instructions in CONTRIBUTING.md in a `make init` task, with a few proposed tweaks to the Ruby environment. This is also an experiment to see if our GitHub actions still work with those changes.

#skip-changelog

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,7 +38,7 @@ With that, the repo is fully set up and you are ready to run all commands.
 
 ## Tests
 
-The's tests depend on our test server. To run the automated tests, you first need to have the server running locally with
+The tests depend on our test server. To run the automated tests, you first need to have the server running locally with
```
Comment (philipphofmann) [inline CONTRIBUTING.md:41]: Good catch thanks 🙏

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ We welcome suggested improvements and bug fixes for `sentry-cocoa`, in the form
 
 ## Setting up an Environment
 
+💡 You might be able to do everything described below by running `make init`.
```
Comment (philipphofmann) [inline CONTRIBUTING.md:14]: `m`: I think we should be more confident here. Why not just write here that you can use `make init` or follow the steps below?

**Review by philipphofmann — COMMENTED** [review]
Awesome thanks @armcknight. 

Please prefix the title of your future PR according to this guideline please https://develop.sentry.dev/code-review/#commit-message-format. I think I should be more clear about that in the Contributing.

I made one minor comment.

File: CONTRIBUTING.md (hunk 2 above)
Comment (armcknight) [reply CONTRIBUTING.md:14]: absolutely, will update this section!

Comment (armcknight) [reply CONTRIBUTING.md]: Please check out my changes in https://github.com/getsentry/sentry-cocoa/pull/1536/commits/97714f0e216597b8c97a5bb013fba4ecc45bfb1c

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,10 @@
+init:
+	wehich brew || /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
Comment (bruno-garcia) [inline Makefile:2]: Did you mean `which`?

```suggestion
	which brew || /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

Comment (bruno-garcia) [reply Makefile]: I'll save up on a review cycle given the time difference and assume the answer is yes. :) Committed

Comment (armcknight) [reply Makefile]: heh thanks :)

**Review by philipphofmann — APPROVED** [review]
LGTM thanks.

**Pull request merged**